target/riscv/gdbstub.c

```diff
@@ -228,6 +228,103 @@ static int riscv_gen_dynamic_csr_xml(CPUState *cs, int base_reg)
     return CSR_TABLE_SIZE;
 }
 
+#if defined(TARGET_CHERI)
+static struct SCR {
+    int index;
+    const char *name;
+    bool code;
+} scrs[] = {
+    { .index = CheriSCR_UTCC, .name = "utcc", .code = true },
+    { .index = CheriSCR_UTDC, .name = "utdc"},
+    { .index = CheriSCR_UScratchC, .name = "uscratchc"},
+    { .index = CheriSCR_UEPCC, .name = "uepcc", .code = true },
+
+    { .index = CheriSCR_STCC, .name = "stcc", .code = true },
+    { .index = CheriSCR_STDC, .name = "stdc"},
+    { .index = CheriSCR_SScratchC, .name = "sscratchc"},
+    { .index = CheriSCR_SEPCC, .name = "sepcc", .code = true },
+
+    { .index = CheriSCR_MTCC, .name = "mtcc", .code = true },
+    { .index = CheriSCR_MTDC, .name = "mtdc"},
+    { .index = CheriSCR_MScratchC, .name = "mscratchc"},
+    { .index = CheriSCR_MEPCC, .name = "mepcc", .code = true },
+};
+
+static inline cap_register_t *get_scr(CPUArchState *env, uint32_t index)
+{
+    switch (index) {
+    case CheriSCR_PCC: return &env->PCC;
+    case CheriSCR_DDC: return &env->DDC;
+
+    case CheriSCR_UTCC: return &env->UTCC;
+    case CheriSCR_UTDC: return &env->UTDC;
+    case CheriSCR_UScratchC: return &env->UScratchC;
+    case CheriSCR_UEPCC: return &env->UEPCC;
+
+    case CheriSCR_STCC: return &env->STCC;
+    case CheriSCR_STDC: return &env->STDC;
+    case CheriSCR_SScratchC: return &env->SScratchC;
+    case CheriSCR_SEPCC: return &env->SEPCC;
+
+    case CheriSCR_MTCC: return &env->MTCC;
+    case CheriSCR_MTDC: return &env->MTDC;
+    case CheriSCR_MScratchC: return &env->MScratchC;
+    case CheriSCR_MEPCC: return &env->MEPCC;
+
+    case CheriSCR_BSTCC: return &env->VSTCC;
```

Comment:
Belong between S and M if including them, but I wouldn't bother, hypervisor bits will only really be done properly for the standardised version. Unless you're reusing the existing function of course, then it's fine for them to stay there.